artichoke-backend/src/class_registry.rs

```diff
@@ -106,7 +77,7 @@ impl ClassRegistry for Artichoke {
             self.with_ffi_boundary(|mrb| {
                 if let Some(mut rclass) = rclass.resolve(mrb) {
                     let value = sys::mrb_obj_new(mrb, rclass.as_mut(), arglen, args.as_ptr());
-                    Some(Value::from(value))
+                    Some(Self::Value::from(value))
```

Comment:
I like using `Self::Value` in the function signatures to match the trait definition, but in the function body, I'd prefer if we use the concrete type instead of indirecting through the trait.

```suggestion
                    Some(Value::from(value))
```